relay-server/src/services/processor.rs

```diff
@@ -192,6 +192,17 @@ impl ProcessingGroup {
             ))
         }
 
+        // Extract user report and user feedback.
+        let report_items = envelope.take_items_by(|item| {
+            matches!(item.ty(), &ItemType::UserReport | &ItemType::ClientReport)
```

Comment:
User reports are items that require event payloads, hence should be processed in the same group as attachments (or standalone attachments). We should follow up with refactoring the standalone attachment group above (really the "doesn't have an event"-group) to cover all items that have `requires_event` set to true. 

Note that the `requires_event` flag means that these items are existentially bound to an event instance in Sentry. That's the reason they are grouped together with events: If the event is dropped, so will be the other items. However, that doesn't mean the event payload must be in the same envelope. We should verify that the  other pipelines handle that correctly.